samtranslator/model/apigateway.py

```diff
@@ -85,3 +86,142 @@ def make_auto_deployable(self, stage, swagger=None):
+
+            if identity_validation_expression:
+                swagger[APIGATEWAY_AUTHORIZER_KEY]['identityValidationExpression'] = identity_validation_expression
+
+        return swagger
+
+    def _construct_role(self, function_invoke_role):
+        return function_invoke_role if function_invoke_role != 'NONE' else None
+
+    def _get_identity_validation_expression(self):
+        return self.identity and self.identity.get('ValidationExpression')
+
+    def _get_identity_source(self):
+        if not self.identity:
+            return None
+
+        if self.identity.get('Headers'):
+            identity_source_headers = map(lambda h: 'method.request.header.' + h, self.identity.get('Headers'))
+
+        if self.identity.get('QueryStrings'):
+            identity_source_query_strings = map(lambda qs: 'method.request.querystring.' + qs, self.identity.get('QueryStrings'))
+
+        if self.identity.get('StageVariables'):
+            identity_source_stage_variables = map(lambda sv: 'stageVariables.' + sv, self.identity.get('StageVariables'))
+
+        if self.identity.get('Context'):
+            identity_source_context = map(lambda c: 'context.' + c, self.identity.get('Context'))
+
+        identity_source_array = identity_source_headers + identity_source_query_strings + identity_source_stage_variables + identity_source_context
+        identity_source = ', '.join(identity_source_array)
+
+        return identity_source
+
+    def _get_user_pool_arn_array(self):
+        return self.user_pool_arn if isinstance(self.user_pool_arn, list) else [self.user_pool_arn]
+
+    def _get_swagger_header_name(self):
+        authorizer_type = self._get_type()
+        payload_type = self._get_function_payload_type()
+
+        if authorizer_type == 'LAMBDA' and payload_type == 'REQUEST':
+            return 'Unused'
+
+        return self._get_identity_header()
+
+    def _get_type(self):
+        if self.user_pool_arn:
+            return 'COGNITO_USER_POOLS'
+
+        return 'LAMBDA'
+
+    def _get_identity_header(self):
+        if not self.identity or not self.identity.get('Header'):
+            return 'Authorization'
+
+        return self.identity.get('Header')
+
+    def _get_reauthorize_every(self):
+        if not self.identity or not self.identity.get('ReauthorizeEvery'):
+            return None
+
+        return self.identity.get('ReauthorizeEvery')
+
+    def _get_function_invoke_role(self):
+        if not self.function_invoke_role or self.function_invoke_role == 'NONE':
+            return None
+
+        return self.function_invoke_role
+
+    def _get_swagger_authtype(self):
+        authorizer_type = self._get_type()
+
+        return 'cognito_user_pools' if authorizer_type == 'COGNITO_USER_POOLS' else 'custom'
+
+    def _get_function_payload_type(self):
+        return 'TOKEN' if not self.function_payload_type else self.function_payload_type
```

Comment:
Add some validation (higher up than this) to check payload type is either TOKEN or REQUEST

Reply:
Done